### PR TITLE
Update examples to use immutability

### PR DESCRIPTION
**Pull request by AndyTitu**

### CODE REVIEW

File: examples/index.cjs [hunk 1]
```diff
@@ -67,28 +68,39 @@ async function manageItems() {
...
+  let newItem = {
+    ...item,
+    fields: item.fields
+      .filter((f) => {
+        return f.id == "password";
+      })
+      .map((f) => {
+        f.value = "new password";
+        return f;
+      }),
```
Comment (Marton6) [inline examples/index.cjs:100]: This is incorrect.

Keep in mind that we're overwriting the item's fields, so using `.filter()` here will mean only the password field it preserved, all other fields are discarded. A correct solution would use `map` to preserve all fields and only change the password field.

File: examples/index.cjs [hunk 2]
```diff
@@ -67,28 +68,39 @@ async function manageItems() {
...
 
-  // Edit an item
-  item.fields[0].value = "other value";
-  let updatedItem = await client.items.put(item);
+  // Edit an item (change the password)
+  let newItem = {
+    ...item,
+    fields: item.fields
+      .filter((f) => {
+        return f.id == "password";
```
Comment (Marton6) [inline examples/index.cjs:95]: I would match the field based on the field name, as most 1Password users are more familiar with field names and a lot less so with field IDs.

File: examples/index.mjs [hunk 3]
```diff
@@ -59,27 +60,37 @@ let item = await client.items.create({
...
+let newItem = {
+  ...item,
+  fields: item.fields
+    .filter((f) => {
+      return f.id == "password";
+    })
+    .map((f) => {
+      f.value = "new password";
+      return f;
+    }),
```
Comment (Marton6) [inline examples/index.mjs:91]: Same feedback here

**Review by Marton6 — CHANGES_REQUESTED** [review]

File: examples/index.cjs (hunk 1 above)
Comment (AndyTitu) [reply examples/index.cjs]: done

**Review by Marton6 — APPROVED** [review]

**Pull request merged**